ledger/accountdb.go

```diff
@@ -1075,72 +2288,376 @@ func accountsPutTotals(tx *sql.Tx, totals ledgercore.AccountTotals, catchpointSt
 	return err
 }
 
-// accountsNewRound updates the accountbase and assetcreators tables by applying the provided deltas to the accounts / creatables.
-// The function returns a persistedAccountData for the modified accounts which can be stored in the base cache.
-func accountsNewRound(tx *sql.Tx, updates compactAccountDeltas, creatables map[basics.CreatableIndex]ledgercore.ModifiedCreatable, proto config.ConsensusParams, lastUpdateRound basics.Round) (updatedAccounts []persistedAccountData, err error) {
+type accountsWriter interface {
+	insertAccount(addr basics.Address, normBalance uint64, data baseAccountData) (rowid int64, err error)
+	deleteAccount(rowid int64) (rowsAffected int64, err error)
+	updateAccount(rowid int64, normBalance uint64, data baseAccountData) (rowsAffected int64, err error)
+
+	insertResource(addrid int64, aidx basics.CreatableIndex, data resourcesData) (rowid int64, err error)
+	deleteResource(addrid int64, aidx basics.CreatableIndex) (rowsAffected int64, err error)
+	updateResource(addrid int64, aidx basics.CreatableIndex, data resourcesData) (rowsAffected int64, err error)
+
+	insertCreatable(cidx basics.CreatableIndex, ctype basics.CreatableType, creator []byte) (rowid int64, err error)
+	deleteCreatable(cidx basics.CreatableIndex, ctype basics.CreatableType) (rowsAffected int64, err error)
+
+	close()
+}
+
+type accountsSQLWriter struct {
+	insertCreatableIdxStmt, deleteCreatableIdxStmt             *sql.Stmt
+	deleteByRowIDStmt, insertStmt, updateStmt                  *sql.Stmt
+	deleteResourceStmt, insertResourceStmt, updateResourceStmt *sql.Stmt
+}
+
+func (w *accountsSQLWriter) close() {
```

Comment:
why not use the same pattern fro closing the statements as `func (qs *accountsDbQueries) close()`